src/components/dashboard/components/dictionary/AddDictionary.jsx

```diff
@@ -2,12 +2,12 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import DictionaryModal from '../dictionary/common/DictionaryModal';
-import { createDictionary, createDictionaryUser, fetchingOrganizations } from '../../../../redux/actions/dictionaries/dictionaryActionCreators';
+import { createDictionary, createDictionaryUser, fetchingOrganizations, fetchDictionaries } from '../../../../redux/actions/dictionaries/dictionaryActionCreators';
 
 export class AddDictionary extends React.Component {
   submit = data => (
     data.owner === 'Individual' ? this.props.createDictionaryUser(data).then(this.props.handleHide)
-      : this.props.createDictionary(data).then(this.props.handleHide))
+      : this.props.createDictionary(data).then(this.props.handleHide)).then(this.props.fetchDictionaries())
```

Comment:
These promises should not be resolved in the component, you should create an action creator (thunk) for this.

Reply:
What do you think about making a new ticket to fix that because that particular file was created by someone else and what I added was their suggestion to update state after creating a new dictionary

Reply:
ok